cmd/query/app/static_handler.go

```diff
@@ -27,35 +27,40 @@ import (
 )
 
 const (
-	staticAssetsRoot = "jaeger-ui-build/build/"
+	defaultStaticAssetsRoot = "jaeger-ui-build/build/"
 )
 
 var (
 	staticRootFiles = []string{"favicon.ico"}
 )
 
 // StaticAssetsHandler handles static assets
-type StaticAssetsHandler struct{}
+type StaticAssetsHandler struct {
+	staticAssetsRoot string
+}
 
 // NewStaticAssetsHandler returns a StaticAssetsHandler
-func NewStaticAssetsHandler() *StaticAssetsHandler {
-	return &StaticAssetsHandler{}
+func NewStaticAssetsHandler(staticAssetsRoot string) *StaticAssetsHandler {
```

Comment:
nit: it seems you don't really need this method. The handler can be created as empty struct, and you can apply the defaulting of the root later.